# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: 05896dbeb434
# Full Hash: 05896dbeb434d6f3ca9324f01eb67020d932f50f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Determine webgl's layers::TextureType in Present. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79617
# ==============================================================================

diff -r 8439f89e170b -r 05896dbeb434 dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Mon Jun 15 18:26:12 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Mon Jun 15 18:26:14 2020 +0000
@@ -402,11 +402,7 @@
   const RefPtr<layers::ImageBridgeChild> imageBridge =
       layers::ImageBridgeChild::GetSingleton();
 
-  auto texType = layers::TextureType::Unknown;
-  if (imageBridge) {
-    texType = layers::PreferredCanvasTextureType(*imageBridge);
-  }
-
+  const auto texType = layers::TexTypeForWebgl(imageBridge);
   Present(nullptr, texType);
 }
 
@@ -426,8 +422,8 @@
 }
 
 Maybe<layers::SurfaceDescriptor> ClientWebGLContext::GetFrontBuffer(
-    WebGLFramebufferJS* const fb, const layers::TextureType type) {
-  return Run<RPROC(GetFrontBuffer)>(fb ? fb->mId : 0, type);
+    WebGLFramebufferJS* const fb) {
+  return Run<RPROC(GetFrontBuffer)>(fb ? fb->mId : 0);
 }
 
 // -